File: tasks/implementations/php/tasks/image_resizing.php
<?php
function run_image_resizing(string $size, string $fixturesRoot): void {
  $tokens = preg_split('/\s+/', trim(file_get_contents($fixturesRoot . "/generated/image/{$size}.ppm")));
  $width = intval($tokens[1]);
  $height = intval($tokens[2]);
  $data = array_map('intval', array_slice($tokens, 4));
  $checksum = 0;
  for ($y = 0; $y < intdiv($height, 2); $y++) {
    for ($x = 0; $x < intdiv($width, 2); $x++) {
      $idx = (($y * 2) * $width + ($x * 2)) * 3;
      $checksum += $data[$idx] + $data[$idx + 1] + $data[$idx + 2];
    }
  }
  echo $checksum . PHP_EOL;
}

File: tasks/implementations/cpp/tasks/image_resizing.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

void run_image_resizing(const std::string& size, const std::string& fixturesRoot) {
    std::ifstream file(fixturesRoot + "/generated/image/" + size + ".ppm");
    std::string magic;
    int width = 0, height = 0, max_value = 0;
    file >> magic >> width >> height >> max_value;
    std::vector<int> data(width * height * 3);
    for (int& value : data) file >> value;
    long long checksum = 0;
    for (int y = 0; y < height / 2; ++y) {
        for (int x = 0; x < width / 2; ++x) {
            int idx = ((y * 2) * width + (x * 2)) * 3;
            checksum += data[idx] + data[idx + 1] + data[idx + 2];
        }
    }
    std::cout << checksum << std::endl;
}
